### PR TITLE
ci(examples): validate app wadm.yaml

### PR DESCRIPTION
**Pull request by ricochet**
This adds an additional step to building example components.

## Feature or Problem
Ensure `wash app validate` passes for example `wadm.yaml`.

## Related Issues

#2697

### Manual Verification

Review of GH CI

### CODE REVIEW

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: .github/workflows/examples-components.yml [hunk 1]
```diff
@@ -168,6 +168,11 @@ jobs:
         run: |
           npm install -g @bytecodealliance/jco
           npm install -g @bytecodealliance/componentize-js
+      # Validate example wadm.yaml if the example has one
+      - name: validate wadm.yaml
+        if: ${{ hashFiles(format('examples/{0}/{1}/components/wadm.yaml', matrix.project.lang, matrix.project.name)) != '' }}
```
Comment (ricochet) [inline .github/workflows/examples-components.yml:173]: Blergh, I can't figure out what to change to make this work. We could opt to require wadm's for every component example. Thoughts?

Comment (ricochet) [reply .github/workflows/examples-components.yml:173]: And also should it switch on local.wadm.yaml vs wadm.yaml?

Comment (vados-cosmonic) [reply .github/workflows/examples-components.yml:173]: Here you can solve it by checking whether the file(s) are there or not in a meta step above! I'll commit something

Comment (vados-cosmonic) [reply .github/workflows/examples-components.yml]: Let me know what you think about this: 

https://github.com/wasmCloud/wasmCloud/pull/2700/commits/8836dac192076f937f82063d755a647b9b4fc49c

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

**Pull request merged**